### PR TITLE
Bump nerdctl to v0.16.1

### LINKED ISSUE #1418: Bump nerdctl to 0.16.1
There are a bunch of fixed related to logging into hosted registries in [v0.16.1](https://github.com/containerd/nerdctl/milestone/12?closed=1).

### PR DESCRIPTION
**Pull request by jandubois**
Fixes various regressions with `nerdctl login`

Fixes #1418

### CODE REVIEW

**Comment by jandubois** [comment]
This PR bump the WSL distro to 0.12.1 instead of 0.13+ to avoid pulling in changes from #1302 

Once this PR has been merged, we should create another bump to 0.14 to pull the new nerdctl into the main branch as well, once the dnsmasq changes have been merged.

File: scripts/download/lima.mjs [hunk 1]
```diff
@@ -7,7 +7,7 @@ import path from 'path';
 import { download, getResource } from '../lib/download.mjs';
 
 const limaRepo = 'https://github.com/rancher-sandbox/lima-and-qemu';
-const limaTag = 'v1.17';
+const limaTag = 'v1.19';
 
 const alpineLimaRepo = 'https://github.com/lima-vm/alpine-lima';
 const alpineLimaTag = 'v0.2.5';
```
Comment (dschunack) [inline scripts/download/lima.mjs:13]: Hi, It's needed to update also the alpineLimaTag to v0.2.6 otherwise the old nerdctl version is used.

Comment (jandubois) [reply scripts/download/lima.mjs:13]: @dschunack Thank you, you are correct! And the change to `limaTag` should be in #1423 and not here...

Comment (jandubois) [reply scripts/download/lima.mjs:13]: Oh, it is already part of #1423, so just needs to be removed here

Comment (jandubois) [reply scripts/download/lima.mjs]: Thank you, should be fixed now!

File: scripts/download/lima.mjs [hunk 2]
```diff
@@ -10,7 +10,7 @@ const limaRepo = 'https://github.com/rancher-sandbox/lima-and-qemu';
 const limaTag = 'v1.17';
 
 const alpineLimaRepo = 'https://github.com/lima-vm/alpine-lima';
-const alpineLimaTag = 'v0.2.5';
+const alpineLimaTag = 'v0.2.6';
```
Comment (mook-as) [inline scripts/download/lima.mjs:13]: > `Error: Error downloading https://github.com/lima-vm/alpine-lima/releases/download/v0.2.6/alpine-lima-rd-3.14.3-x86_64.iso.sha512sum`

I think we forgot to actually ship 0.2.6? https://github.com/lima-vm/alpine-lima/releases/tag/v0.2.6 doesn't have any assets (other than the source tarballs).

**Review by mook-as — CHANGES_REQUESTED** [review]
Don't we need matching changes in `lima.ts` and `wsl.ts` to actually use those images?

**Comment by jandubois** [comment]
> Don't we need matching changes in `lima.ts` and `wsl.ts` to actually use those images?

We do, thanks for catching that!

Can we restructure the code so that hard-coded versions only need to be updated in a single place?

**Review by mook-as — APPROVED** [review]

**Pull request merged**